File: server-microservice/chapter-service/src/main/java/com/lms/chapterservice/model/Chapter.java
package com.lms.chapterservice.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.util.Date;
import java.util.List;
import java.util.UUID;

@AllArgsConstructor
@NoArgsConstructor
@Data
@JsonInclude
@Entity
@Table(name = "Chapter")
public class Chapter {
    @Id
    private String id = UUID.randomUUID().toString();
    private String title;
    private String description;
    private String videoUrl;
    private Integer position;
    private Boolean isPublished = false;
    private Boolean isFree = false;

    private String courseId;

    @CreationTimestamp
    private Date createdAt;
    @UpdateTimestamp
    private Date updatedAt;
}

File: nextjs-client/types/Chapter.ts
import { UserProgress } from './UserProgress';

interface Chapter {
    id: string;
    title: string;
    description: string;
    videoUrl: string;
    position: number;
    isPublished: boolean;
    isFree: boolean;
    courseId: string;
    muxDataId: string;
    userProgress: UserProgress[];
    createdAt: Date;
    updatedAt: Date;
}

export type { Chapter };